examples/step-98/step-98.cc

```diff
@@ -0,0 +1,975 @@
+    //
+    // Micro-benchmarking can help determine the optimal method for
+    // distributing DoFs, as performance depends on several factors. For
+    // instance, with Clang, lookup tables are highly efficient; the overhead
+    // of copying data between cell vectors and patch vectors is minimal,
+    // comparable to a standard copy operation of the same data size. With GCC,
+    // an index-computing implementation (not included here) outperforms lookup
+    // tables, though it's still significantly slower than Clang. Using
+    // `DistributorLookup` is crucial for efficiency, as it provides the
+    // mapping between patch DoFs and cell DoFs.
+    using PatchEval =
+      FEPatchEvaluation<FEEval,
+                        PatchDistributors::DistributorLookup<dim, fe_degree>>;
+
+
+
+    // Set up the FEPatchEvaluation object. It takes the patch storage
+    // and a configured FEEvaluation object (based on the MatrixFree data
+    // associated with the patch storage).
+    PatchEval patch_eval(patch_storage,
+                         FEEval(*patch_storage.get_matrix_free()));
+
+    // Determine the start, end, and step direction for the patch loop based
+    // on whether it's a forward or backward sweep.
+    const auto begin = do_forward ? patch_range.first : patch_range.second - 1;
+    const auto end   = do_forward ? patch_range.second : patch_range.first - 1;
+    const auto step  = do_forward ? 1 : -1;
+
+
+    // Iterate over the assigned range of patches.
+    for (auto patch_index = begin; patch_index != end; patch_index += step)
+      {
+        // Reinitialize the patch evaluation helper for the current patch index.
+        // This sets up internal structures for the specific cells in this
+        // patch.
+        patch_eval.reinit(patch_index);
+
+        // Allocate aligned vectors for local computations on the patch.
+        // The size is determined by the number of unique DoFs within the patch.
+        AlignedVector<number> local_rhs(patch_eval.n_patch_dofs());
+        AlignedVector<number> local_residual(patch_eval.n_patch_dofs());
+        AlignedVector<number> local_correction(patch_eval.n_patch_dofs());
+
+        // --- Step 1: Gather the right-hand side (RHS) for the patch ---
+        // Read the relevant global RHS values ('src') into the FEEvaluation
+        // objects associated with the cells of the current patch.
+        patch_eval.read_dof_values(src);
+        // Gather the values from the individual cells' FEEvaluation objects
+        // into the contiguous 'local_rhs' vector for the patch.
+        // 'false' indicates this is a simple gather, we are not accumulating
+        // results In other words, values that are share between cell will be
+        // imported once,only for the cell with lowest index (cells are ordered
+        // lexicographically).
+        patch_eval.gather_local_to_patch(ArrayView<double>(local_rhs), false);
```

Comment:
@kronbichler the flag false is not self-explanatory and this code without a comment is hard to read. I had to look up documentation a couple of times to while working on my own code. I think there should be an enum to specify that explicitely, can you suggest a good naming? The same applies distribute_patch_to_local.